### PR TITLE
test: hive db engine spec

### PR DESCRIPTION
**Pull request by kstrz**
### SUMMARY
increase superset/db_engine_specs/hive.py tests coverage

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kstrz** [comment]
Hey! Could you look at it? @willbarrett @dpgaspar

File: tests/db_engine_specs/hive_tests.py [hunk 1]
```diff
@@ -247,3 +265,83 @@ def is_readonly(sql: str) -> bool:
...
+@mock.patch("superset.db_engine_specs.base.BaseEngineSpec.fetch_data")
+def test_fetch_data_success(fetch_data_mock):
+    return_value = ["a", "b"]
+    fetch_data_mock.return_value = return_value
+    cursor = mock.Mock()
+    assert HiveEngineSpec.fetch_data(cursor) == return_value
+
+
+@mock.patch("superset.db_engine_specs.hive.HiveEngineSpec._latest_partition_from_df")
+def test_where_latest_partition(mock_method):
```
Comment (dpgaspar) [inline tests/db_engine_specs/hive_tests.py:335]: to increase coverage would be nice to add 2 further tests:
- `latest_partition` raises an exception
- when `values` is None and `columns` is None

File: tests/db_engine_specs/hive_tests.py [hunk 2]
```diff
@@ -168,6 +169,23 @@ def test_create_table_from_csv_append() -> None:
         )
 
 
+@mock.patch(
+    "superset.db_engine_specs.hive.config",
+    {**app.config, "CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC": lambda *args: True},
+)
+@mock.patch("superset.db_engine_specs.hive.g", spec={})
+@mock.patch("tableschema.Table")
+def test_create_table_from_csv_if_exists_fail(mock_table, mock_g):
```
Comment (dpgaspar) [inline tests/db_engine_specs/hive_tests.py:178]: Too increase coverage, add one test with `Table.schema` also to cover: https://github.com/apache/superset/blob/master/superset/db_engine_specs/hive.py#L235

File: tests/db_engine_specs/hive_tests.py [hunk 3]
```diff
@@ -168,6 +169,23 @@ def test_create_table_from_csv_append() -> None:
...
+    mock_table.infer.return_value = {}
+    mock_g.user = True
+    mock_database = mock.MagicMock()
+    mock_database.get_df.return_value.empty = False
+    with pytest.raises(SupersetException, match="Table already exists"):
+        HiveEngineSpec.create_table_from_csv(
+            "foo.csv", Table("foobar"), mock_database, {}, {"if_exists": "fail"}
```
Comment (dpgaspar) [inline tests/db_engine_specs/hive_tests.py:185]: add one further test with `{"if_exists": "replace"}` would be great

**Review by dpgaspar — APPROVED** [review]
Looks good, would be nice to add a couple more tests

**Review by willbarrett — APPROVED** [review]

**Pull request merged**